llvm/lib/Target/SPIRV/SPIRVPreTranslationLegalizer.cpp

```diff
@@ -107,12 +106,16 @@ SPIRVPreTranslationLegalizer::processFunctionSignature(Function *F) {
   auto *ThisFuncMD = MDNode::get(B.getContext(), MDArgs);
   FuncMD->addOperand(ThisFuncMD);
 
-  for (auto *U : F->users()) {
+  // Make a list of all uses, then modify the instructions, otherwise the uses
```

Comment:
Confusion between uses/users. Plus you may try to use `make_early_inc_range` instead of a temporary array

Reply:
Thanks, `make_early_inc_range` works great, I'll fix this place.